Closes #3804   Move WebP Compatibility option to One-Click Rocket Add-on

mostafa-hisham opened the pull request:
## Description

Move WebP Compatibility option to One-Click Rocket Add-on

Fixes #3804 

## Type of change


- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Is the solution different from the one proposed during the grooming?
No

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] test locally see if the media option moved to addon ta

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

mostafa-hisham committed:
move webp to add os

remyperona commented on inc/Engine/Admin/Settings/Page.php:
```diff
@@ -878,20 +879,14 @@ private function assets_section() {
-		} else {
-			$imagify_link = '<a href="https://wordpress.org/plugins/imagify/" target="_blank" rel="noopener noreferrer">';
-		}
+		/*$webp_beacon      = $this->beacon->get_suggest( 'webp' );*/
```
You can completely delete the line instead of commenting it

mostafa-hisham replied:
Done

remyperona commented on inc/Engine/Admin/Settings/Page.php:
```diff
@@ -878,20 +879,14 @@ private function assets_section() {
 			[
 				'title'            => __( 'Media', 'rocket' ),
-				'menu_description' => __( 'LazyLoad, embeds, WebP', 'rocket' ),
+				'menu_description' => __( 'LazyLoad, embeds', 'rocket' ),
```
```suggestion
				'menu_description' => __( 'LazyLoad, image dimensions', 'rocket' ),
```

Since we're removing also the embeds option, we can update the description to add the image dimensions option too which was not mentioned before

mostafa-hisham replied:
Done

remyperona commented on inc/Engine/Admin/Settings/Page.php:
```diff
@@ -1851,6 +1818,56 @@ private function addons_section() {
 			);
 		}
 
+		if ( apply_filters( 'rocket_display_webp_options_tab', true ) ) {
```
```suggestion
```

We don't need a filter to hide/show this block

remyperona commented on inc/Engine/Admin/Settings/Page.php:
```diff
@@ -1851,6 +1818,56 @@ private function addons_section() {
+							),
+				]
+			);
+		}
```
```suggestion
```

remyperona commented on inc/Engine/Admin/Settings/Page.php:
```diff
@@ -1851,6 +1818,56 @@ private function addons_section() {
+								'page'              => 'addons',
+								'settings_page'     => 'webp',
+								'default'           => 0,
+								'helper'            => 'assassasd',
```
```suggestion
```

mostafa-hisham replied:
Fixed

remyperona commented on inc/Engine/Admin/Settings/Page.php:
```diff
@@ -1851,6 +1818,56 @@ private function addons_section() {
+					/**
+					 * Add more content to the 'cache_webp' setting field.
+					 *
+					 * @since  3.9.1
```
```suggestion
					 * @since  3.4
```

mostafa-hisham replied:
Fixed

remyperona commented on inc/classes/subscriber/Media/class-webp-subscriber.php:
```diff
@@ -266,11 +266,9 @@ public function webp_section_description( $cache_webp_field ) {
-				'<a href="' . esc_url( $webp_beacon['url'] ) . '" data-beacon-article="' . esc_attr( $webp_beacon['id'] ) . '" target="_blank" rel="noopener noreferrer">',
-				'</a>'
-			);
+				esc_html( _n( 'You are using %1$s to serve WebP images so you do not need to enable this option. If you prefer to have WP Rocket serve WebP for you instead, please disable them from serving in %1$s.', 'You are using %1$s to serve WebP images so you do not need to enable this option. If you prefer to have WP Rocket serve WebP for you instead, please disable them from serving in %1$s.', count(  ...
+				esc_html( wp_sprintf_l( '%l', $serving ) )
+				);
```
What's the reason to remove the beacon link on all those helper?

mostafa-hisham replied:
I added  Beacon links back

but there is a problem
this `class-webp-subscriber.php` is adding a warning block  to the "webp checkox field" but the  "one-click addon" field does not have a warning block in it 
but we display a helper message similar to the one in warning in the "one-click addon"  field helper.

should I add the warning block in "one-click addon"  field and edit the js to make "one-click addon" `wpr-radio-ui` (sliding checkbox button)  work with the warning message button?

In my opinion
we should leave "one-click addon"  fields without warning block
, remove warning edit  from `class-webp-subscriber.php`
and maybe edit the helper message a little bit to tell the user that "wp-rocket didn't detect any compatible WebP plugin"

https://github.com/wp-media/wp-rocket/blob/723d6c0181b7b3849c3e15bd60cdb2e0233a38f8/inc/classes/subscriber/Media/class-webp-subscriber.php#L353-L363

remyperona replied:
I agree with you we can adapt the code for what exists in the one-click addon section

mostafa-hisham committed:
add correct beacon link and remove dispaly filter

mostafa-hisham committed:
add beacon links back

mostafa-hisham committed:
remove warning and efit helper message

mostafa-hisham committed:
fix return type

mostafa-hisham committed:
fix addon css

mostafa-hisham committed:
fix admin css conflict merge with branch-3.10

mostafa-hisham committed:
fix webp unit test

mostafa-hisham committed:
fix delay js test case

mostafa-hisham committed:
phpcs error fix

mostafa-hisham commented:
@Tabrisrp 
I removed the warning and edit the helper text please have a look at it

also for some reason after merging with brach-3.10 to fix admin CSS conflict 
in`tests/Fixtures/inc/Engine/Optimization/DelayJS/Subscriber/addDelayJsScript.php`
the integration test said that `$expected `is not a string after I removed the `$expected `value and add it again it worked fine.

Commit:
Merge branch 'branch-3.10' into enhancement/3804-move-webp-to-add-on

piotrbak commented:
@GeekPress Is this expected UI? I think the red/grey messages are kind of duplicating what we want to tell the customers:
![image](https://user-images.githubusercontent.com/26767037/133443850-90ee3b41-b25e-4d07-a8b4-08a90204164a.png)

mostafa-hisham commented:
@piotrbak 
can you hard clear the browser cache?
the style should look like this

![Screenshot from 2021-09-15 15-53-31](https://user-images.githubusercontent.com/29744252/133446378-b3c99742-f472-4c62-931a-de6b3ce94ba4.png)

piotrbak commented:
@mostafa-hisham The line-height problem is still there, I'm thinking about the descriptions.

mostafa-hisham committed:
fix line height issues

mostafa-hisham committed:
Merge branch 'enhancement/3804-move-webp-to-add-on' of github.com:wp-media/wp-rocket into enhancement/3804-move-webp-to-add-on

piotrbak commented:
@mostafa-hisham When Imagify is used and set to serve the WebP images, we're changing the font color to show users that it's not possible to enable it. However, we're still able to enable the add-on itself:
![image](https://user-images.githubusercontent.com/26767037/133467135-6ae1abcf-5d7f-48c5-8ddc-9555c6950b43.png)

It wasn't possible in previous approach

mostafa-hisham committed:
fix styling

mostafa-hisham committed:
add on-click-addon disabled style

mostafa-hisham committed:
phpcs fix

mostafa-hisham commented:
@piotrbak on click add-on fields can now be disabled please have a look

piotrbak commented:
@mostafa-hisham Switching animation doesn't work now when enabling and Imagify is not enabled:
https://jmp.sh/MKGPcZA

The ajax request is sent and the button is enabled after the refhres though.

mostafa-hisham committed:
fix disabled css

mostafa-hisham committed:
push licence-data.php

mostafa-hisham committed:
Merge remote-tracking branch 'origin/branch-3.10' into enhancement/3804-move-webp-to-add-on

mostafa-hisham committed:
merge with branch 3.10 to fix css conflict

piotrbak commented:
@mostafa-hisham after importing settings with WebP caching on, when Imagify is set to show the WebP, we'll have greyed out setting until saving settings. Steps to reproduce:
1. Imagify is disabled
2. Enable WebP caching and export settings
3. Enable Imagify and set it to serve WebP
4. Check the WP Rocket add-ons, WebP is disabled
5. Import settings from step 2 and recheck the add-ons, WebP is set to `on`
6. Save WP Rocket settings and recheck the add-ons, WebP is set to `off`

Can we somehow guard against it?

mostafa-hisham committed:
fix wepb add-on stuck on off

mostafa-hisham committed:
fix tests

mostafa-hisham commented:
@piotrbak i updated the PR to fix the problem please have a look

piotrbak reviewed (approved):
I think we're good here @mostafa-hisham 
[testrail-report-262.pdf](https://github.com/wp-media/wp-rocket/files/7187394/testrail-report-262.pdf)

@Tabrisrp before going live, we need to take care of messaging. There are 4 different scenarios to cover:
1. Overall description (I think we're good here, it's provided by Jo in the user story)
2. When no image optimization plugins are detected, reminder about the purpose of this add-on
3. When image optimization plugin(s) is/are enabled, but not serving the WebP
4. When image optimization plugin(s) is/are serving the WebP and our option is greyed out

remyperona commented:
I'm going to merge this, and we can discuss the messaging with the support team with an alpha version in their hands
